tempodb/encoding/vparquet/wal_block.go

```diff
@@ -446,6 +446,9 @@ func (b *walBlock) Iterator() (common.Iterator, error) {
 }
 
 func (b *walBlock) Clear() error {
+	if b.file != nil {
+		b.file.Close()
```

Comment:
This is the thing that I think may fix the original issue. b.writer is not closed too, but as far as I see, that it is based on the file, so there is no reason to double close

Reply:
Here, I debugged test `TestWalDropsZeroLength`. Not sure whether the test is simulating real-world workload, but the file is still open when we call `CutCompleteTraces`

![image](https://user-images.githubusercontent.com/3115913/221702293-98c32102-ea93-4bc9-b5d0-e74e1256f476.png)

Reply:
i definitely agree with this change, but let's check and return the error from `b.file.Close()`.

should we run `os.RemoveAll()` even if `b.file.Close()` fails? it might be the safer thing to do

Reply:
Added error aggregation. I agree, I think it's safer to remove everything rather than not, so did that.